pkgs/top-level/all-packages.nix

```diff
@@ -12101,11 +12101,11 @@ with pkgs;
 
   pinentry = libsForQt5.callPackage ../tools/security/pinentry { };
 
-  pinentry-curses = (lib.getOutput "curses" pinentry);
-  pinentry-emacs = (lib.getOutput "emacs" pinentry);
-  pinentry-gtk2 = (lib.getOutput "gtk2" pinentry);
-  pinentry-qt = (lib.getOutput "qt" pinentry);
-  pinentry-gnome = (lib.getOutput "gnome3" pinentry);
+  pinentry-curses = pinentry.override { enabledFlavors = [ "curses" ]; };
+  pinentry-emacs = pinentry.override { enabledFlavors = [ "emacs" ]; };
+  pinentry-gtk2 = pinentry.override { enabledFlavors = [ "gtk2" ]; };
+  pinentry-qt = pinentry.override { enabledFlavors = [ "qt" ]; };
+  pinentry-gnome3 = pinentry.override { enabledFlavors = [ "gnome3" ]; };
```

Comment:
I vaguely recall that curses was supposed to be a fallback, if gui could not be started. But now the only reference I found is https://github.com/NixOS/nixpkgs/pull/71095#discussion_r334303343, which is not very informative.